### ISSUE TITLE
trying to use same class twice by inheritance: works here, not there

### ISSUE THREAD

**Issue by kapouer** [ungap/custom-elements-builtin#15]
```
class HTMLElementMedia extends HTMLMediaElement {
  /* ... */
}
HTMLCustomElement.define('element-video', class HTMLElementVideo extends HTMLElementMedia {}, 'video');
HTMLCustomElement.define('element-audio', class HTMLElementAudio extends HTMLElementMedia {}, 'audio');
```
Upon instantiation, it fails with
```
Uncaught TypeError: Illegal constructor
    at new HTMLElementMedia (media.js:1)
    at new <anonymous> (media.js:46)
```

It fails with chrome 80 / firefox 72 but works in webkitgtk 2.26 (where builtins need polyfill).

I suppose there is no obvious solution (besides doing the inheritance "by hand").
Is there ? Maybe it should be documented, since it might work with your polyfill and then break with native builtins.

**Comment by WebReflection** [comment]
my poly doesn't touch Chrome and FF, so I'm not sure why there is a bug here, but also you need to pass `{extends: 'video'}`, not just `video` or `audio`.

The constructor has well described caveats too, but that's for Safari WebKit only.

TL;DR your code is not standard, so it's expected to fail where it fails, but I also don't think I should documented anything, as you're using the API wrong, for what I can see.

**Comment by kapouer** [comment]
In fact in this case extending video builtin requires extending HTMLVideoElement. Likewise for HTMLAudioElement.
This has nothing to do with your module, sorry.
